connectors/sources/graphql.py

```diff
@@ -0,0 +1,512 @@
+                raise
+        except Exception:
+            raise
+
+    async def get(self, graphql_query):
+        params = {"query": graphql_query}
+        async with self.session.get(url=self.url, params=params) as response:
+            json_response = await response.json()
+            if not json_response.get("errors"):
+                data = json_response.get("data", {})
+                return data
+            msg = f"Error while executing query. Exception: {json_response['errors']}"
+            raise Exception(msg)
+
+    async def post(self, graphql_query):
+        """Invoke GraphQL request to fetch response.
+
+        Args:
+            graphql_query (string): GraphQL query
+
+        Raises:
+            UnauthorizedException: Unauthorized exception.
+            ClientResponseError: ClientResponseError exception.
+            exception: An instance of an exception class.
+
+        Yields:
+            dictionary: Client response
+        """
+        query_data = {
+            "query": graphql_query,
+            "variables": self.variables,
+        }
+        async with self.session.post(url=self.url, json=query_data) as response:
+            json_response = await response.json()
+            if not json_response.get("errors"):
+                data = json_response.get("data", {})
+                return data
+            msg = f"Error while executing query. Exception: {json_response['errors']}"
+            raise Exception(msg)
+
+    async def close(self):
+        self._sleeps.cancel()
+        await self.session.close()
+        del self.session
+
+    async def ping(self):
+        await self.make_request(graphql_query=PING_QUERY)
+
+
+class GraphQLDataSource(BaseDataSource):
+    """GraphQL"""
+
+    name = "GraphQL"
+    service_type = "graphql"
+
+    def __init__(self, configuration):
+        """Setup the connection to the GraphQL instance.
+
+        Args:
+            configuration (DataSourceConfiguration): Instance of DataSourceConfiguration class.
+        """
+        super().__init__(configuration=configuration)
+        self.graphql_client = GraphQLClient(configuration=configuration)
+
+    def _set_internal_logger(self):
+        self.graphql_client.set_logger(self._logger)
+
+    @classmethod
+    def get_default_configuration(cls):
+        """Get the default configuration for GraphQL.
+
+        Returns:
+            dict: Default configuration.
+        """
+        return {
+            "http_endpoint": {
+                "label": "HTTP URL & Endpoint",
+                "order": 1,
+                "type": "str",
+            },
+            "http_method": {
+                "display": "dropdown",
+                "label": "GET/POST",
+                "options": [
+                    {"label": "GET", "value": "get"},
+                    {"label": "POST", "value": "post"},
+                ],
+                "order": 2,
+                "type": "str",
+                "value": "POST",
```

Comment:
```suggestion
                "value": "post",
```
We should match the value, not the label